TestModels/dafny-dependencies/StandardLibrary/runtimes/python/src/standard_library/extern/UTF8.py

```diff
@@ -0,0 +1,86 @@
+from standard_library.internal_generated_dafny.UTF8 import *
+import standard_library.internal_generated_dafny.UTF8
+import _dafny
+import struct
+
+'''
+Extern UTF8 encode and decode methods.
+
+Note:
+Python's `.encode('utf-8')` does not handle surrogates.
+These methods are expected to handle surrogates (e.g. "\uD808\uDC00").
+To work around this, we encode Dafny strings into UTF-16-LE (little endian)
+and decode them before encoding into UTF-8 (`_strict_tostring`).
+To decode, we reverse the encode step. (`_reverse_strict_tostring`)
+'''
+
+# Extend the Dafny-generated class with our extern methods
+class default__(standard_library.internal_generated_dafny.UTF8.default__):
+
+  @staticmethod
+  def Encode(s):
+    try:
+      return Wrappers.Result_Success(_dafny.Seq(
+          default__._strict_tostring(s).encode('utf-8')
+      ))
+    # Catch both Encode and Decode errors.
+    # The `try` block involves both encoding and decoding.
+    except (UnicodeEncodeError, UnicodeDecodeError):
+      return Wrappers.Result_Failure(_dafny.Seq("Could not encode input to Dafny Bytes."))
+
+  @staticmethod
+  def _strict_tostring(s):
+    '''
+    Converts a Dafny Seq of ASCII characters
+    into a string by writing a UTF-16-LE decoding,
```

Comment:
From above, I assume this is a typo:
```suggestion
    into a string by writing a UTF-16-LE encoding,
```